document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -2433,6 +2461,71 @@ void testSameHierarchyUpdateThrowsException(final String datastoreName) throws I
               collection.bulkUpdate(
                   query, updates, UpdateOptions.builder().returnDocumentType(NONE).build()));
     }
+
+    @ParameterizedTest
+    @ArgumentsSource(AllProvider.class)
+    void testAddOperatorThrowExceptionForNonNumericValue(final String datastoreName)
+        throws IOException {
+      final Collection collection = getCollection(datastoreName, UPDATABLE_COLLECTION_NAME);
+      createCollectionData("query/updatable_collection_data.json", UPDATABLE_COLLECTION_NAME);
+
+      // assert exception for string
+      final SubDocumentUpdate addString =
+          SubDocumentUpdate.builder()
+              .subDocument("item")
+              .operator(ADD)
+              .subDocumentValue(SubDocumentValue.of("Comb"))
+              .build();
+
+      final Query query = Query.builder().build();
+      final List<SubDocumentUpdate> updates = List.of(addString);
+      assertExceptionForNonNumericValues(collection, datastoreName, query, updates);
+
+      // assert exception for list
+      final SubDocumentUpdate addList =
+          SubDocumentUpdate.builder()
+              .subDocument("props.added.list")
+              .operator(ADD)
+              .subDocumentValue(SubDocumentValue.of(new Integer[] {5, 1, 5}))
+              .build();
+      final Query query_addList = Query.builder().build();
+      final List<SubDocumentUpdate> updates_addList = List.of(addList);
+      assertExceptionForNonNumericValues(collection, datastoreName, query_addList, updates_addList);
+
+      // assert exception for Object
+      final SubDocumentUpdate addObject =
+          SubDocumentUpdate.builder()
+              .subDocument("props.newObject")
+              .operator(ADD)
+              .subDocumentValue(
+                  SubDocumentValue.of(
+                      new Document[] {
+                        new JSONDocument(Map.of("name", "Pluto")),
+                        new JSONDocument(Map.of("name", "Mars"))
+                      }))
+              .build();
+      final Query query_addObject = Query.builder().build();
+      final List<SubDocumentUpdate> updates_addObject = List.of(addObject);
+      assertExceptionForNonNumericValues(
+          collection, datastoreName, query_addObject, updates_addObject);
+    }
+
+    private void assertExceptionForNonNumericValues(
+        Collection collection, String datastoreName, Query query, List<SubDocumentUpdate> updates) {
+      if (MONGO_STORE.equals(datastoreName)) {
+        assertThrows(
```

Comment:
Any reason why "IOException" can't be thrown for MongoDB?
The consumers of the library might not know (and hence might not catch) MongoWriteException.

If it's not possible, probably, we need to fix it to throw IOException.